src/engraving/libmscore/slur.cpp

```diff
@@ -366,8 +366,44 @@ void SlurSegment::adjustEndpoints()
     }
 }
 
+Shape SlurSegment::getSegmentShape(Segment* seg, ChordRest* startCR, ChordRest* endCR)
+{
+    staff_idx_t startStaffIdx = startCR->staffIdx();
+    Shape segShape = seg->staffShape(startStaffIdx).translated(seg->pos() + seg->measure()->pos());
+    // If cross-staff, also add the shape of second staff
+    if (slur()->isCrossStaff() && seg != startCR->segment()) {
+        staff_idx_t endStaffIdx = (endCR->staffIdx() != startStaffIdx) ? endCR->staffIdx() : endCR->vStaffIdx();
+        SysStaff* startStaff = system()->staves().at(startStaffIdx);
+        SysStaff* endStaff = system()->staves().at(endStaffIdx);
+        double dist = endStaff->y() - startStaff->y();
+        Shape secondStaffShape;
+        secondStaffShape.add(seg->staffShape(endStaffIdx).translated(seg->pos() + seg->measure()->pos()));
+        secondStaffShape.translate(PointF(0.0, dist));
```

Comment:
Why not 
```c++
Shape secondStaffShape = seg->staffShape(endStaffIdx).translated(seg->pos() + seg->measure()->pos()));
secondStaffShape.translate(PointF(0.0, dist));
```
or even 
```c++
Shape secondStaffShape = seg->staffShape(endStaffIdx).translated(seg->pos() + seg->measure()->pos() + PointF(0.0, dist)));
```
?

Reply:
👍 Though I prefer the first option, as I'd still rather keep the vertical translation (which is the tricky one cause it depends on staff distance) visually separate from the horizontal.